federation/pkg/federation-controller/sync/controller.go

```diff
@@ -505,5 +516,23 @@ func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*
 			})
 		}
 	}
+
+	for _, cluster := range unselectedClusters {
```

Comment:
Yeah, I hear you. FWIW, The old clusters slice is the union of the new selected and unselected slices, so there's no additional iteration occuring, it's just being split into two separate loops.